File: WebClient/src/app/content/posts/dtos/post.document.ts
export interface PostDocument {
    id: string
    authorId: string
    authorName: string
    authorSpecialization: string
    authorProfilePictureUrl: string
    communityId?: string | null;
    communityName?: string | null;
    communityLogoUrl?: string | null;
    title: string
    content: string
    createdAt: Date
    updatedAt: Date
    lastInteractedAt: Date | null;
    mediaCount: number;
    firstMediaUrl: string | null;
    tags: string[];
    commentsCount: number;
    reactionsCount: number;
    viewsCount: number;
}

File: backend/SNS.Domain/Search/Documents/PostDocument.cs
namespace SNS.Domain.Search.Documents;

public class PostDocument
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public List<string> Topics { get; set; } = new List<string>();
    public List<string> Tags { get; set; } = new List<string>();
}